lib/puppet/util/platform.rb

```diff
@@ -17,6 +17,24 @@ def default_paths
         %w{/usr/sbin /sbin}
       end
       module_function :default_paths
+
+      def fips_enabled?
+        # We currently only plan to support FIPS on RHEL7
+        # so we try to eliminate anything un-supported using what we have available
+        # in lieu of not being able to specifically check for it.
+
+        return false if Puppet::Util::Platform.windows?
+
+        fips_status_file = '/proc/sys/crypto/fips_enabled'
+
+        begin
+           File.open(fips_status_file, &:readline)[0].chr == '1'
```

Comment:
You can shorten this to just `File.read(fips_status_file, 1)`, which will only read and return only the first byte. Also recommend making `fips_status_file` a frozen constant.